org.eclipse.xtext.builder/src/org/eclipse/xtext/builder/resourceloader/ParallelResourceLoader.java

```diff
@@ -185,7 +185,7 @@ public void run() {
 
 				// push resource to the queue, wait if queue is full
 				try {
-					resourceQueue.put(Tuples.create(uri, resource, exception));
+					getResourceQueue().put(Tuples.create(uri, resource, exception));
```

Comment:
am not sure if this has performance impact if called 10ks of times

Reply:
Good question. My gut feeling is that hotspot will inline the field access.